seacrowd/sea_datasets/fleurs/fleurs.py

```diff
@@ -0,0 +1,291 @@
+    if len(CONFIG_SUFFIXES_FOR_TASK) != len(_SUPPORTED_TASKS):
+        raise AssertionError("Config prefixes doesn't matched in terms of `len` with `_SUPPORTED_TASKS`!")
+    if len(CONFIG_SUFFIXES_FOR_TASK) == 0:
+        raise AssertionError("Config prefixes and `_SUPPORTED_TASKS` have `len` of 0!")
+
+
+conform_init_config()
+
+
+def construct_configs_on_langs(languages: list = None) -> List[SEACrowdConfig]:
+    """
+    The function `construct_configs` constructs a list of SEACrowdConfig objects based on the provided
+    languages or a default language, and returns the list.
+
+    input:
+        languages (list): The `languages` parameter is a list that specifies the languages for which the
+        configurations need to be constructed. If no languages are provided (value=None), the first value in language config
+        will be used.
+    output:
+        a list of `SEACrowdConfig` objects based on instantiated init variables
+    """
+    # set output var
+    config_list = []
+
+    # set flag whether the task is lang-agnostic based on extended `_SUPPORTED_TASKS`
+    IS_TASK_LANG_SUBSETTED = [True, False]
+
+    TASKS_AND_CONFIG_SUFFIX_PAIRS = list(zip(_SUPPORTED_TASKS, CONFIG_SUFFIXES_FOR_TASK, IS_TASK_LANG_SUBSETTED))
+
+    # implement source schema
+    version, config_name_prefix = _SOURCE_VERSION, "source"
+    config_list += [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_{_LANG}_{config_name_prefix}",
+            version=datasets.Version(version),
+            description=f"{_DATASETNAME} {config_name_prefix} schema for language code {_LANG}",
+            schema=f"{config_name_prefix}",
+            subset_id=_LANG,
+        )
+        for _LANG in languages
+    ]
+
+    # implement SEACrowd schema
+    version, config_name_prefix = _SEACROWD_VERSION, "seacrowd"
+    for (task_obj, config_name_suffix, is_lang_subsetted) in TASKS_AND_CONFIG_SUFFIX_PAIRS:
+        if is_lang_subsetted:
+            # construct configs based on its lang, since the task & config needs to defined per lang
+            # for this dataloader, Tasks.SPEECH_RECOGNITION will enter this condition
+            config_list += [
+                SEACrowdConfig(
+                    name=f"{_DATASETNAME}_{_LANG}_{config_name_prefix}_{config_name_suffix}",
+                    version=datasets.Version(version),
+                    description=f"{_DATASETNAME} {config_name_prefix} schema for {task_obj.name} and language code {_LANG}",
+                    schema=f"{config_name_prefix}_{config_name_suffix}",
+                    subset_id=_LANG,
+                )
+                for _LANG in languages
+            ]
+
+        else:
+            # else, its defined for all languages
+            # for this dataloader, Tasks.SPEECH_LANGUAGE_IDENTIFICATION will enter this condition
+            config_list.append(
+                SEACrowdConfig(
+                    name=f"{_DATASETNAME}_all_{config_name_prefix}_{config_name_suffix}",
```

Comment:
Hi @sabilmakbar, LGTM!! Could you use `fleurs_seacrowd_speech` instead of `fleurs_seacrowd_all_speech`?

Reply:
Okay, but will it make sense to re-implement `fleurs_source` again if this is changed?

cc @holylovenia @SamuelCahyawijaya